src/redux/sagas/utils/expenditures.ts

```diff
@@ -160,3 +175,124 @@ export const getPayoutsWithSlotIds = (
     slotId: index + 1,
   }));
 };
+
+type PayoutWithSlotId = ExpenditurePayout & {
+  slotId: number;
+};
+
+export const getImmediatelyClaimableSlots = (slots: ExpenditureSlot[]) => {
+  return slots.filter(
+    (slot) =>
+      slot.claimDelay !== null &&
+      slot.claimDelay !== undefined &&
+      slot.claimDelay === 0,
+  );
+};
+
+export const getPayoutsWithSlotIdsFromSlots = (
+  claimableSlots: ExpenditureSlot[],
+): PayoutWithSlotId[] => {
+  return claimableSlots.flatMap((slot) =>
+    (slot.payouts || []).map((payout) => ({
+      ...payout,
+      slotId: slot.id,
+    })),
+  );
+};
+
+interface ClaimExpendituresPayoutsParams {
+  colonyAddress: Address;
+  nativeExpenditureId: number;
+  claimablePayouts: PayoutWithSlotId[];
+  annotationMessage?: string;
+  metaId: string;
+}
+
+const getPayoutChannelId = (payout: PayoutWithSlotId) =>
+  `${payout.slotId}-${payout.tokenAddress}`;
+
+// NOTE: this is called from 3 sagas so it's designed to be wrapped in a try catch
+export function* claimExpenditurePayouts({
+  colonyAddress,
+  claimablePayouts,
+  nativeExpenditureId,
+  annotationMessage,
+  metaId,
+}: ClaimExpendituresPayoutsParams) {
+  if (claimablePayouts.length === 0) {
+    return;
+  }
+
+  const batchKey = 'claimExpenditurePayouts';
+
+  const { annotatePayoutChannel, ...channels } =
+    yield createTransactionChannels(metaId, [
+      ...claimablePayouts.map(getPayoutChannelId),
+      'annotatePayoutChannel',
+    ]);
+
+  // Create one claim transaction for each slot
+  yield all(
+    claimablePayouts.map((payout, index) =>
+      fork(createTransaction, channels[getPayoutChannelId(payout)].id, {
+        context: ClientType.ColonyClient,
+        methodName: 'claimExpenditurePayout',
+        identifier: colonyAddress,
+        params: [nativeExpenditureId, payout.slotId, payout.tokenAddress],
+        group: {
+          key: batchKey,
+          id: metaId,
+          index,
+        },
+        ready: false,
+      }),
+    ),
+  );
+  if (annotationMessage) {
+    yield fork(createTransaction, annotatePayoutChannel.id, {
+      context: ClientType.ColonyClient,
+      methodName: 'annotateTransaction',
+      identifier: colonyAddress,
+      group: {
+        key: batchKey,
+        id: metaId,
+        index: claimablePayouts.length,
+      },
+      ready: false,
+    });
+  }
+
+  for (const payout of claimablePayouts) {
+    const payoutChannelId = getPayoutChannelId(payout);
+
+    yield takeFrom(
+      channels[payoutChannelId].channel,
+      ActionTypes.TRANSACTION_CREATED,
+    );
+    if (annotationMessage) {
+      yield takeFrom(
+        annotatePayoutChannel.channel,
+        ActionTypes.TRANSACTION_CREATED,
+      );
+    }
```

Comment:
This breaks the saga if you try add annotation to it, you've only got one annotation tx but are trying to wait for its results multiple times.

I've actually removed annotations from this saga in #1922, so feel free to do that as well